### DISCUSSION TITLE
Check if model satisfies constraints

### DISCUSSION THREAD

**Discussion by egolf-cs** [Z3Prover/z3#6856] [Q&A]
What is the proper way to check if a model, m, satisfies the constraints of a Solver object?

One way to do this check:
1. Construct constraint c that is satisfied only by m
2. push
3. add constraint c
4. check if sat
5. pop
6. m satisfies the constraints iff the check in 4 was sat

Is this the best way to do this?

**Comment by NikolajBjorner** [comment]
- you can sometimes use the eval function on models to simplify constraints using a model
